crates/swift-bridge-ir/src/bridged_type/bridgeable_result.rs

```diff
@@ -441,11 +444,25 @@ typedef struct {c_enum_name}{{{c_tag_name} tag; union {c_fields_name} payload;}}
         let ok = self.ok_ty.to_swift_type(type_pos, types);
         let err = self.err_ty.to_swift_type(type_pos, types);
 
+        let (ok_val, err_val, condition) = if self.ok_ty.can_be_encoded_with_zero_bytes() {
+            (
+                ok,
+                format!("{err}(ptr: rustFnRetVal!)"),
+                "rustFnRetVal == nil",
+            )
+        } else {
+            (
+                format!("{ok}(ptr: rustFnRetVal.ok_or_err!)"),
+                format!("{err}(ptr: rustFnRetVal.ok_or_err!)"),
+                "rustFnRetVal.is_ok",
+            )
+        };
+
         format!(
-            r#"if rustFnRetVal.is_ok {{
-        wrapper.cb(.success({ok}(ptr: rustFnRetVal.ok_or_err!)))
+            r#"if {condition} {{
+        wrapper.cb(.success({ok_val}))
     }} else {{
-        wrapper.cb(.failure({err}(ptr: rustFnRetVal.ok_or_err!)))
+        wrapper.cb(.failure({err_val}))
     }}"#
```

Comment:
Let's add a codegen test since we're changing codegen.

Mainly because without a robust codegen test suite diving into future problems / implementing future optimizations would become a nightmare where a bunch of integration tests fail and it's hard to investigate why.

And especially since this is one of the most common Results that we'd expect users to use in the wild.

We can just add a `-> Result<(), OpaqueRustType>` codegen test to the Result codegen tests.

https://github.com/chinedufn/swift-bridge/blob/20baa731361c9cb86c2b4a9f5c7d0a5f65f6ff45/crates/swift-bridge-ir/src/codegen/codegen_tests/result_codegen_tests.rs#L6-L5

There are plenty of example async codegen tests here https://github.com/chinedufn/swift-bridge/blob/2d60d9ee83a153b07c5413af9f9cc3ad6075c9f4/crates/swift-bridge-ir/src/codegen/codegen_tests/async_function_codegen_tests.rs#L182-L268